resources/assets/js/features/trip/components/Containers/EditTripContainer.js

```diff
@@ -36,19 +36,26 @@ class EditTripContainer extends React.Component {
     componentDidMount() {
         EditTripService.getTrip(this.props.id)
             .then(response => {
-                response = EditTripService.transformData(response);
-                const routes = response.routes[0];
+                response = EditTripService.transformData(response.data);
+
+                const routes = response.routes.data;
+
                 this.setState({
                     trip: response,
                     startPoint: {
-                        address: routes.from.formatted_address
+                        address: routes[0].from.formatted_address
                     },
                     endPoint: {
-                        address: routes.to.formatted_address
+                        address: routes[routes.length - 1].to.formatted_address
                     },
                 });
+
                 this.onSelectStartPoint(this.state.startPoint.address);
                 this.onSelectEndPoint(this.state.endPoint.address);
+
+                setTimeout(() => {
+                    this.props.addWaypointsFromRoutes(this.state.trip.routes.data);
+                }, 10);
```

Comment:
if you want to execute some method in other scope, 0 will be enough, otherwise - it could make bad joke with you